### PR TITLE
Share y-axis limits on GOES plots

### PR DESCRIPTION
**Pull request by dstansby**
Fixes https://github.com/sunpy/sunpy/issues/6163. A bit of a hacky fix, but as far as I can tell the only way to share limits in Matplotlib without having the same locator/formatter on both Axis instances.

### CODE REVIEW

**Comment by hayesla** [comment]
I wonder could we simplify this part of the code completely and now use twinx but rather just ax.text it, for example, if we remove the `ax2 = axes.twinx()` part to label the GOES classes we could do:
```python
    values = [2.5e-8, 2.5e-7, 2.5e-6, 2.5e-5, 2.5e-4] # the 2.5 to plot in centre
    labels = ['A', 'B', 'C', 'M', 'X']
    for value, label in zip(values, labels):
        axes.text(1.02, value, label, transform=axes.get_yaxis_transform())
```

and seems to work well:
example_goes

and solves the issue with the ylims form #6163 (i.e. `ax.set_ylim(1e-7,1e-5)`)

example_goes_zoom

thoughts?

**Comment by dstansby** [comment]
A cunning plan! That's much better, I've implemented it.

File: changelog/6486.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+The right-hand y-axis of GOES timeseries plots with labelled flare classes
```
Comment (hayesla) [inline changelog/6486.bugfix.rst:1]:
```suggestion
The right-hand y-axis of the GOES-XRS timeseries plots with labelled flare classes
```

**Review by hayesla — APPROVED** [review]

**Comment by dstansby** [comment]
Merging as a bugfix with one approval - I only updated the figure hashes since the positive review.

**Pull request merged**